plugins/nodes/src/js/data/MesosMasters.js

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { Observable } from "rxjs/Observable";
+import "rxjs/add/observable/timer";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/retry";
+import "rxjs/add/operator/filter";
+
+import MesosStateStore from "#SRC/js/stores/MesosStateStore";
+
+import LeaderGrid from "../components/LeaderGrid";
+
+function isEmptyObject(obj) {
+  return Object.keys(obj).length === 0;
+}
+
+function hostPort(address) {
+  return `${address.hostname}:${address.port}`;
+}
+
+function supportsRegion(master) {
+  return (
+    master.domain &&
+    master.domain.fault_domain &&
+    master.domain.fault_domain.region &&
+    master.domain.fault_domain.region.name
+  );
+}
+
+export function getRegion(master) {
+  return supportsRegion(master)
+    ? master.domain.fault_domain.region.name
+    : "N/A";
+}
+
+const STORE_POLL_INTERVAL = 2000;
+
+function mesosMasterEmpty() {
+  return {
+    hostPort: undefined,
+    version: undefined,
+    electedTime: undefined,
+    startTime: undefined,
+    region: undefined
+  };
+}
+
+export function mesosMasterInfo(
+  masterDataSource,
```

Comment:
Typescript would have told me this is a function - can you add JSDoc comments? 😇

Reply:
Can you teach me? I do not know JSDoc.